routing/cross_mwm_index_graph.hpp

```diff
@@ -155,6 +164,48 @@ class CrossMwmIndexGraph final
   }
 
 private:
+  std::vector<m2::PointD> GetFeaturePointsBySegment(Segment const & segment)
+  {
+    auto const & handle = m_dataSource.GetMwmHandleByCountryFile(m_numMwmIds->GetFile(segment.GetMwmId()));
+    if (!handle.IsAlive())
+      return {};
+
+    auto const & mwmId = handle.GetId();
+
+    auto const & featureId = FeatureID(mwmId, segment.GetFeatureId());
+    std::vector<m2::PointD> geometry;
+
+    auto const fillGeometry = [&geometry](FeatureType & ftype)
+    {
+      ftype.ParseGeometry(FeatureType::BEST_GEOMETRY);
+      geometry.reserve(ftype.GetPointsCount());
+      for (uint32_t i = 0; i < ftype.GetPointsCount(); ++i)
+        geometry.emplace_back(ftype.GetPoint(i));
+    };
+
+    m_dataSource.ReadFeature(fillGeometry, featureId);
+
+    return geometry;
+  }
+
+  /// \brief Check segment for equality point by point.
```

Comment:
`/// \brief Checks segment for equality point by point.`

Reply:
done